### PR TITLE
#158626108 Export Workouts in JSON Format

### PR DESCRIPTION
**Pull request by tonykimathi**
#### What does this PR do?
Gives the user the ability to export workouts as a JSON file. 
#### Description of Task to be completed?
Previously, wger only supported transferring a user's workout schedule in PDF format which cannot be used by other applications.
The export workout as JSON feature would be used to allow the user to export their workouts and send them to other users so they can add them in their workout schedules.
#### How should this be manually tested?
- Follow the instructions on the README file on how to set up the application locally
- Login into the system
- Navigate to the workouts tab and select workouts on the dropdown menu
- Select your workout and on the options button to your right, select `Export workout as JSON` on the drop down menu
#### Any background context you want to provide?
- The file will be downloaded, and will be available in the `Downloads` folder
#### What are the relevant pivotal tracker stories?
[#158626108](https://www.pivotaltracker.com/story/show/158626108)
#### Screenshots
image
image
image
image

### CODE REVIEW

**Comment by parseen254** [comment]
@tonykimathi please make the file downloadable. The user has no right to check or view the wger dir so once they export the file it should be presented to them as a download.

**Comment by tonykimathi** [comment]
Hi @parseendavid. Thanks for your recommendations. I've applied them and updated the PR accordingly. Please have a look.

**Comment by parseen254** [comment]
screen shot 2018-07-19 at 8 53 10 am
@tonykimathi I am getting an error when there are no workouts to export. Handle this edge case.

**Comment by parseen254** [comment]
and rebase with develop @tonykimathi

**Comment by tonykimathi** [comment]
@parseendavid On it.

**Comment by tonykimathi** [comment]
@parseendavid Fixed. Kindly review the changes made.

**Comment by parseen254** [comment]
screen shot 2018-07-19 at 3 11 55 pm
I am still having issues with exporting. Check how download feature, why are trying to open a possibly none existing file ?

**Comment by tonykimathi** [comment]
@parseendavid I think I've fixed what was causing the error and handled other edge cases. Please review and let me know if this task is complete.

File: wger/manager/views/workout.py [hunk 1]
```diff
@@ -208,6 +212,73 @@ def add(request):
...
+            set_list = []
+
+            for set_exercise in set_exercises:
+                set_info = {}
+                list_of_exercises = []
+                exercise_info = {}
+
+                for info in set_exercise['exercise_list']:
+                    exercise_info['exercise'] = info['obj'].name
+                    exercise_info['set_number'] = info['setting_list']
```
Comment (parseen254) [inline wger/manager/views/workout.py]: @tonykimathi why a list called set_number?

**Review by parseen254 — CHANGES_REQUESTED** [review]

File: wger/manager/views/workout.py (hunk 1 above)
Comment (tonykimathi) [reply wger/manager/views/workout.py]: @parseendavid I agree that the naming could be different. I've updated it. See changes.

**Review by parseen254 — APPROVED** [review]

**Pull request merged**